test/Dialect/FIRRTL/lower-types.mlir

```diff
@@ -1057,36 +1057,117 @@ firrtl.module private @is1436_FOO() {
+    // CHECK:   %2 = firrtl.ref.resolve %sink_data : !firrtl.ref<uint<64>>
+    %w = firrtl.wire   : !firrtl.bundle<valid: uint<1>, ready: uint<1>, data: uint<64>>
+    // CHECK:   %w_valid = firrtl.wire   : !firrtl.uint<1>
+    // CHECK:   %w_ready = firrtl.wire   : !firrtl.uint<1>
+    // CHECK:   %w_data = firrtl.wire   : !firrtl.uint<64>
+    firrtl.strictconnect %w, %0 : !firrtl.bundle<valid: uint<1>, ready: uint<1>, data: uint<64>>
+    // CHECK:   firrtl.strictconnect %w_valid, %0 : !firrtl.uint<1>
+    // CHECK:   firrtl.strictconnect %w_ready, %1 : !firrtl.uint<1>
+    // CHECK:   firrtl.strictconnect %w_data, %2 : !firrtl.uint<64>
+  }
+  firrtl.module private @RefTypeVectors1(in %a: !firrtl.ref<vector<uint<1>, 2>>, out %b: !firrtl.vector<uint<1>, 2>) {
+    // CHECK-LABEL: firrtl.module private @RefTypeVectors1(
+    // CHECK-SAME: in %a_0: !firrtl.ref<uint<1>>, in %a_1: !firrtl.ref<uint<1>>, out %b_0: !firrtl.uint<1>, out %b_1: !firrtl.uint<1>) {
+    %0 = firrtl.ref.resolve %a : !firrtl.ref<vector<uint<1>, 2>>
+    // CHECK:  %0 = firrtl.ref.resolve %a_0 : !firrtl.ref<uint<1>>
+    // CHECK:  %1 = firrtl.ref.resolve %a_1 : !firrtl.ref<uint<1>>
+    firrtl.strictconnect %b, %0 : !firrtl.vector<uint<1>, 2>
+    // CHECK:  firrtl.strictconnect %b_0, %0 : !firrtl.uint<1>
+    // CHECK:  firrtl.strictconnect %b_1, %1 : !firrtl.uint<1>
+  }
+  firrtl.module private @RefTypeBundles2() {
+    %x = firrtl.wire   : !firrtl.bundle<a: uint<1>, b: uint<2>>
+    %0 = firrtl.ref.send %x : !firrtl.bundle<a: uint<1>, b: uint<2>>
+    // CHECK:   %0 = firrtl.ref.send %x_a : !firrtl.uint<1>
+    // CHECK:   %1 = firrtl.ref.send %x_b : !firrtl.uint<2>
+    %1 = firrtl.ref.resolve %0 : !firrtl.ref<bundle<a: uint<1>, b: uint<2>>>
+    // CHECK:   %2 = firrtl.ref.resolve %0 : !firrtl.ref<uint<1>>
+    // CHECK:   %3 = firrtl.ref.resolve %1 : !firrtl.ref<uint<2>>
+  }
+  firrtl.module private @RefTypeVectors(out %c: !firrtl.vector<uint<1>, 4>) {
+    %x = firrtl.wire   : !firrtl.vector<uint<1>, 4>
+    %0 = firrtl.ref.send %x : !firrtl.vector<uint<1>, 4>
+    // CHECK:  %0 = firrtl.ref.send %x_0 : !firrtl.uint<1>
+    // CHECK:  %1 = firrtl.ref.send %x_1 : !firrtl.uint<1>
+    // CHECK:  %2 = firrtl.ref.send %x_2 : !firrtl.uint<1>
+    // CHECK:  %3 = firrtl.ref.send %x_3 : !firrtl.uint<1>
+    %1 = firrtl.ref.resolve %0 : !firrtl.ref<vector<uint<1>, 4>>
+    // CHECK:  %4 = firrtl.ref.resolve %0 : !firrtl.ref<uint<1>>
+    // CHECK:  %5 = firrtl.ref.resolve %1 : !firrtl.ref<uint<1>>
+    // CHECK:  %6 = firrtl.ref.resolve %2 : !firrtl.ref<uint<1>>
+    // CHECK:  %7 = firrtl.ref.resolve %3 : !firrtl.ref<uint<1>>
+    firrtl.strictconnect %c, %1 : !firrtl.vector<uint<1>, 4>
+    // CHECK:  firrtl.strictconnect %c_0, %4 : !firrtl.uint<1>
+    // CHECK:  firrtl.strictconnect %c_1, %5 : !firrtl.uint<1>
+    // CHECK:  firrtl.strictconnect %c_2, %6 : !firrtl.uint<1>
+    // CHECK:  firrtl.strictconnect %c_3, %7 : !firrtl.uint<1>
```

Comment:
These tests feel a bit redundant. Cant his be reduced to two tests: one that checks ref type vectors and one that checks ref type bundles?